### DISCUSSION TITLE
A paid subscription even on your own hosting?

### DISCUSSION THREAD

**Discussion by NSGrid** [rustdesk/rustdesk-server#297] [General]
Hello! Could you please clarify a point? And why, if I install RustDesk server (version pro - docker pull rustdesk/rustdesk-server-pro-s6) on my device (in particular, I host as a docker-container on my Synology NAS), then I still ask for a license? If this project was originally conceived as a source code project, then why, have some of its features become payable, even if you host the server on your own host? I accept that some features may be paid for, but usually this always applies to features that are hosted in the software developer's cloud. But here, it turns out, to use the professional features even on your own hosting, you have to pay a subscription? That is, there isn't even a one-off payment option. I'm sorry, but this seems to me a very strange approach by the developer.

Also to add, I'm sorry, but it's completely incomprehensible to me why I have to pay for the address book feature, and pay by way of subscription, by hosting a RustDesk server on my personal hosting?

**Comment by dinger1986** [comment]
The opensource client and server is and always will be free and opensource, there is a paid for version, if projects don't get funding they die. 

There's nothing at all to stop someone developing their own api and open sourcing it.

**Comment by NSGrid** [comment]
So that doesn't stop you from selling your product for commercial use to various companies and individuals who are in business. I understand about the financial support, but, you know, I really miss the address book support even for purely personal non-commercial use. I would expect a server that you host to have it. You could think of some option for non-commercial use of not only the basic functions, there was such a theme, for example, with Team Viewer, and there was no possibility to use your own server. Or at least some variant without a paid subscription, but with a one-time payment.

**Comment by dinger1986** [comment]
People still use non-commercial for commercial use.

Putting this into perspective $10 a month is maximum 3 drinks in a bar or a packet of cigarettes

**Comment by NSGrid** [comment]
It's understandable that they use it...
But for me personally I just miss the address book in the free version and that's it, as well as the online/offline client status function and the ability to turn on the computer by WOL directly from the address book.
As for the cost, for Russia, where I live, it's quite a lot, considering the ever-increasing dollar exchange rate, which is already approaching 100 rubles per dollar. In addition, your cost is specified for a month, it means it must be multiplied by 12, so it is for a year. For instance, for personal use I would be ready to pay for Pro version 10-20 dollars per year, but not per month :(
I would also like to know from what country you, dear developer, where are your main activities now, because nowhere on the site does it say, where are you from?

P. S. By the way, on your site in the section of purchasing licenses (https://rustdesk.com/pricing.html), if you click on the button "buy now", it redirects to some page with an error.  This happens both when selecting the Personal plan, and when selecting the Business plan.

**Comment by dinger1986** [comment]
Sorry it does work out so expensive, I hope that the ruble vs the dollar comes down soon :)

> P. S. By the way, on your site in the section of purchasing licenses (https://rustdesk.com/pricing.html), if you click on the button "buy now", it redirects to some page with an error. This happens both when selecting the Personal plan, and when selecting the Business plan.

Thanks, yes the payment portal hasnt been completed quite yet.

**Comment by rustdesk** [comment]
Ready

**Comment by mooleshacat** [comment]
> People still use non-commercial for commercial use.

Only problem I have with this statement, is why I left TeamViewer, and eventually Anydesk. They claimed I was a business, without actually verifying with me that I am actually personal, and then blocked and banned me from use without paying for a business license.

Sure I have like 50 computers in my apartment at times, and I have a grandmother and mother I try to help with theirs. But I am not a business by far.

My concern is this will happen with RustDesk as well. I do not run a business, my usage is entirely personal.

I also agree with the OP it is a weird way to license your software - it is atypical. Usually anything I self host that is open source is free to self host. Example: Nextcloud. I am sure there are many others. Paid service is usually hosted services.

**Comment by dinger1986** [comment]
You do know that statement was a reply to someone and shouldn't be taken out of context?

It can't go the same way as anydesk and teamviewer as the code is out there already.

**Comment by paspo** [comment]
> I also agree with the OP it is a weird way to license your software - it is atypical. Usually anything I self host that is open source is free to self host. Example: Nextcloud. I am sure there are many others. Paid service is usually hosted services.

This is not always the case.
Take for example Bitwarden (the password manager): if you self-host their server solution, a proper license is still required.
And that is the reason why projects like Vaultwarden are widely used.

**Reply by NSGrid** [reply]
I have a Synology NAS running Docker Container Bitwarden (docker pull vaultwarden/server) and no license was required, I use everything fine without any restrictions on my own hosting.

**Reply by dinger1986** [reply]
> I have a Synology NAS running Docker Container Bitwarden (docker pull vaultwarden/server) and no license was required, I use everything fine without any restrictions on my own hosting.

Vaultwarden is (and I quote from their GitHub) Unofficial Bitwarden compatible server written in Rust, formerly known as bitwarden_rs

**Reply by NSGrid** [reply]
Well, in fact, the RustDesk software is just something like that, but the developer still believes that you should charge money, and on a subscription basis, for the address book function (and some other Pro-functions), if even the user hosts it. My opinion hasn't changed, if hosted on my personal hosting - it should be free, at most I would be ready for something like a one-time donation. Well, and some kind of limit on the number of address book contacts, for example, no more than 50 items. For most non-commercial and home use it would be enough, well, and those who have thousands of contacts there, well ok, let it be a corporate use and with the appropriate subscription fee.

**Reply by dinger1986** [reply]
Its not at all the same or similar.

RustDesk Server was always closed until a year or so ago and then it was opensourced. 

Only parts of Bitwarden is opensourced and someone has decided to create their own part which is opensource (Vaultwarden).

There is nothing to stop someone developing their own opensource API server for RustDesk which then would be similar to vaultwarden/bitwarden.

If theres limits on an addressbook people will complain cause they dont need 50 they need 60 and so on, literally impossible.

**Reply by paspo** [reply]
> Well, in fact, the RustDesk software is just something like that, but the developer still believes that you should charge money

In fact, in terms of intellectual property, the developer is the owner of the code, and has the right to decide which license and terms to use for distribution. 

If I am a developer of a cool thing, I can decide to use a standard license or I can do something fancy like requiring "ius primae noctis" on user's daughters (nobody in their mind will accept these terms, but I can request it).

The developer can choose the license for the software he/she creates, and if the user wants to use that software, he/she must agree to that license's terms. No way around that.

Obviously license decision will affect how the project is seen by the community, but that's another story.

> My opinion hasn't changed, if hosted on my personal hosting - it should be free

Why?
First of all, the "free" in "free software" is not meant to be "free as a beer", but more like "free speech".

If your reasoning is "I pay for the my hosting, I don't want to pay for your hosting", than this is missing a big part: all the work in producing the software (and everything is connected to that).

In the life of an open-source project, the incoming money is used to "feed the devs"; without the money, you won't have developers. Any serious (non-hobbyist) open source project needs full time developers, to give users features/updates in a timely manner.
You can't have 8hr/day of developer time if you're not paying. This line makes the difference between an hobbyist project (a "side-project", a "pet project", it has many names) and something than has a relevance can grow in the future.

**Reply by Extarys** [reply]
What I find sad is that RustDesk is more expensive than AnyDesk and I need to host it. (At least for me in Canada.)

**Reply by rustdesk** [reply]
You should compare with AnyDesk's self-hosted pricing https://github.com/rustdesk/rustdesk/discussions/7201#discussioncomment-8531253

**Comment by pricerc** [comment]
> What I find sad is that RustDesk is more expensive than AnyDesk and I need to host it. (At least for me in Canada.)

You must get special pricing in Canada :).

I suppose if you're a single user doing self-hosting it "kind-of" is. But that's not fair comparison.

With AnyDesk, for USD15 / month, you can have (according to their web site), for one licensed user, up to 3 "login" devices, but only one outgoing connection (i.e. you can only use one of those three devices at a time; they may not be enforcing it, but that's what the license says). And 100 "Unattended"/"Managed" devices that you can connect to. Go to one of the "Pro" versions, and each outgoing connection costs you USD30 / month. Have four team members who need to help customer simultaneously? That's $120/month.

With RustDesk; _for free_, you can have as many connections as you like. You just don't get "user management".

But for USD 20/month, you can have basic user management with up to 10 users and 100 "managed" devices. And it scales at $10 for every 10 users, or $1 for 10 additional  "managed" devices.

Ok, the hosting has an additional cost, but if you're hosting RustDesk for use by a limited group of people, then that cost should be negligible. I host mine on the cheapest ARM Linux host offered by Azure. So far works out at less than $10 per month. I use Azure because I get a monthly credit that would otherwise be going to waste. I believe there are even "free" options from e.g., AWS.

And if you have a fixed IP address for your Internet connection, then you can host it on a Raspberry Pi.

**Comment by GuillaumeHullin** [comment]
I would love a one-time purchase option for the address book feature, **but** I also recognize that this specific feature should not be taken lightly. 

Call me paranoid, but the reason I use my password manager (Bitwarden self-hosted with paid license, by the way) as my RustDesk address book is that I do not yet feel confident leaving a service on the Internet with remote access to my other computer protected only by a simple user login + MFA. If compromised... effectively giving away all the keys to the kingdom.

So, I understand the decision to make it a subscription feature. First, because it's more desirable than it is essential for the basic functions (= remote control), and second, because it is a feature that needs to be really well implemented to avoid becoming a security risk for everyone.

**Comment by Tekwizz** [comment]
Since I exclusively use VNC, I only need occasional use of another solution (AnyDesk) currently seems to work well if I lose the uPNP port forwarding when I can't forward a router. If you are getting paid to do remote support as a business, $10/mo is nothing obviously if you need that functionality.
